pkg/controller/filter.go

```diff
@@ -43,11 +44,12 @@ func (f *MastershipFilter) Accept(id types.ID) bool {
 	if err != nil {
 		return false
 	}
-	master, err := f.Store.IsMaster(device)
-	if err != nil {
-		return false
+	master, err := f.Store.GetMastership(device)
+	if err == nil && master != nil {
```

Comment:
@kuujo 
Please make sure it makes sense to you.

Reply:
This should be checking whether the local node is the master. This check merely checks whether some master exists.